### DISCUSSION TITLE
Include "info" dict in create_table statement

### DISCUSSION THREAD

**Discussion by BayerSe** [sqlalchemy/alembic#1328] [Usage Questions] [answered]
Hi,

our Oracle db requires partitions for some of the tables, which we used to add using alembic with the following steps. These are based on the proposal of @zzzeek [here](https://groups.google.com/g/sqlalchemy/c/qCQFD2LNyTQ) (for nicer formatting see [here](https://stackoverflow.com/a/45705933/3072050)). 

It's been a long time since I needed a change in the schema, so I only now realized that there seems to have been a regression / change in behavior.

Main code:
```python
from sqlalchemy import MetaData, Column, String
from sqlalchemy.ext.declarative import declarative_base

metadata = MetaData()
Base = declarative_base(metadata=metadata)


class Table(Base):
    __tablename__ = 'Table'
    name = Column(String(10), primary_key=True)
    __table_args__ = {'info': {'oracle_partition': "PARTITION BY ..."}}
```

This rewriter is added as `process_revision_directives` of `context.configure`  in `env.py` with the goal of adding the `info` dict to the `create_table` statement:

```python
writer = rewriter.Rewriter()

@writer.rewrites(ops.CreateTableOp)
def add_info_to_table(context, revision, op):
    op.kw["info"] = op.columns[0].table.info
    return ops.CreateTableOp(
            table_name=op.table_name,
            columns=op.columns,
            schema=op.schema,
            _namespace_metadata=op._namespace_metadata,
            _constraints_included=op._constraints_included,
            **op.kw,
        )
```

We add this to the migration scripts, which adds the partition command to the created statement: 
```python
import sqlalchemy as sa
from alembic import op
import textwrap

@sa.ext.compiler.compiles(sa.schema.CreateTable, "oracle")
def _add_suffixes(element, compiler, **kw):  # type: ignore
    text = compiler.visit_create_table(element, **kw)
    if "oracle_partition" in element.element.info:
        text += textwrap.dedent(element.element.info["oracle_partition"]).strip()
    return text
```

For `alembic==1.6.5` and `sqlalchemy==1.3.24` this result in the following auto-generated update command:

```python
op.create_table(
    "Table",
    sa.Column("name", sa.String(length=10), nullable=False),
    sa.PrimaryKeyConstraint("name"),
    info={"oracle_partition": "PARTITION BY ..."},
)
```

However, for `alembic==1.12.0` and `sqlalchemy==2.0.22`, the `info` is not added anymore, i.e. I just get

```python
op.create_table(
    "Table",
    sa.Column("name", sa.String(length=10), nullable=False),
    sa.PrimaryKeyConstraint("name"),
)
```

I checked version `alembic==1.7.0` as well, it broke between `1.6.5` and `1.7.0`. In the changelog I found [this](https://alembic.sqlalchemy.org/en/latest/changelog.html#change-4dd6bde9d2096da0af4c803fd1096420) comment, could that be related? 

Thanks and regards
Sebastian

**Comment by zzzeek** [comment]
well at the very least there's a typing error with Rewriter being sent to process_revision_directives

**Comment by zzzeek** [comment]
IIUC the bug here is an autogenerate rendering issue.   rendering "info" overall though doesnt seem like something we would have been doing in any case, it can have anything inside of it that we would not know how to render.

are you sure this used to render "info" at some point?

**Reply by zzzeek** [reply]
hmm looks like the code shoudl be doing it though.  strange

**Reply by BayerSe** [reply]
Thanks for the quick response! Yes I'm sure it works for version 1.6.5, but not for 1.7.0.

**Reply by zzzeek** [reply, marked as answer]
oh yes I guess 1ce816684f213ae50287fb22c would have reversed this then since it's popping it.

issue is in #1329

**Reply by BayerSe** [reply]
Thanks!
